### PR TITLE
1305 project permission flow

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Re-style RequestProjectAccess Modal
- Refactor OverlayDiv & SidePanel to be functional components
- Re-styled no data message on dashboards
---

### Screenshots:
Screenshots can be found in the issue.

### CODE REVIEW

File: packages/web-frontend/src/components/View/NoDataMessage.js [hunk 1]
```diff
@@ -29,9 +38,16 @@ export const NoDataMessage = ({ viewContent }) => {
...
 };
 
 NoDataMessage.propTypes = {
   viewContent: PropTypes.shape(VIEW_CONTENT_SHAPE).isRequired,
+  severity: PropTypes.string,
+  AlertComponent: PropTypes.elementType,
+};
+
+NoDataMessage.defaultProps = {
+  AlertComponent: StyledAlert,
```
Comment (kael89) [inline packages/web-frontend/src/components/View/NoDataMessage.js:51]: Clever!

File: packages/web-frontend/src/components/View/NoDataMessage.js [hunk 2]
```diff
@@ -29,9 +38,16 @@ export const NoDataMessage = ({ viewContent }) => {
...
   }
 
-  return <StyledAlert severity="info">{message}</StyledAlert>;
+  return <AlertComponent severity={severity}>{message}</AlertComponent>;
 };
 
 NoDataMessage.propTypes = {
   viewContent: PropTypes.shape(VIEW_CONTENT_SHAPE).isRequired,
+  severity: PropTypes.string,
+  AlertComponent: PropTypes.elementType,
```
Comment (kael89) [inline packages/web-frontend/src/components/View/NoDataMessage.js:47]: `NoDataMessage`, where `PlainNoDataMessage` is passed in for `AlertComponent`. We could consider

* Making this the default value here and delete `PlainNoDataMessage`
* Do not support passing a component in, but always use `PlainNoDataMessage`

But I'm fine if you think that the existing pattern is a good component abstraction practice

**Review by kael89 — APPROVED** [review]
Excellent work!!

File: packages/web-frontend/src/components/View/NoDataMessage.js (hunk 2 above)
Comment (tcaiger) [reply packages/web-frontend/src/components/View/NoDataMessage.js:47]: Thanks Kostas! well picked up. I will the second option and make it as simple as possible.

**Pull request merged**